scenario2/experiment_orchestrator.py

```diff
@@ -0,0 +1,130 @@
+
+    rendered = template.render(server_args=server_args,
+                            client_args=client_args,
+                            job_name=job_name)
+    print(rendered)
+    resource_dict = yaml.safe_load(rendered)
+    job = Job(resource_dict)
+    job.create()
+
+    return job_name
+
+def wait_for_server(model: str):
+    """Wait for vLLM server to be ready"""
+    url = "http://localhost:8000/v1/models"
+    max_attempts = 50  # 60 seconds total
+
+    print("Waiting for server to be ready...")
+    for attempt in range(max_attempts):
+        try:
+            response = requests.get(url, timeout=5)
+            if model in str(response.json()):
+                print("Server is ready")
+                return True
+        except requests.exceptions.RequestException:
+            pass
+
+        time.sleep(5)
+
+    print("Server failed to start within timeout")
+    return False
+
+def run_experiment(model, mode, remote_exp_folder: str):
+    config_file = f"scenario2_config_{mode}.yaml"
+
+    with open(config_file, "r") as f:
+       full_config = yaml.safe_load(f) # read necessary configs and seed files
+
+    # Set up K8s client
+    config.load_kube_config()
+
+    try:
+        c = Configuration().get_default_copy()
+        c.verify_ssl = False
+    except AttributeError:
+        c = Configuration()
+        c.assert_hostname = False
+        c.verify_ssl = False
+    Configuration.set_default(c)
+
+    # Run the workload
+    print(f"\n{'='*50}")
+    print(f"STARTING SCENARIO 1 benchmark in mode = {mode}")
+    print(f"{'='*50}")
+
+    experiment_configs = full_config["experiments"]
+    for experiment in experiment_configs:
+        chunk_size = experiment["chunk_size"]
+        job_name = start_vllm_server_client(experiment, remote_exp_folder, mode, model, chunk_size)
+        print(f"Created pod '{job_name}'")
+
+def main():
+    modes = ["train", "test"]
+    # models = ["mistralai/Mistral-7B-Instruct-v0.1", "google/gemma-7b", "meta-llama/Llama-3.1-8B","ibm-granite/granite-3.3-8b-instruct", "mistralai/Mistral-Small-24B-Instruct-2501", "Qwen/Qwen3-32B"]
+    # models = ["ibm-granite/granite-3.3-8b-instruct", "mistralai/Mistral-Small-24B-Instruct-2501"]
+    # models = ["Qwen/Qwen2.5-1.5B"]
+    models = ["Qwen/Qwen2.5-0.5B", "Qwen/Qwen2.5-1.5B", "Qwen/Qwen2.5-3B", "Qwen/Qwen2.5-7B", "mistralai/Mistral-7B-Instruct-v0.1", "google/gemma-7b", "meta-llama/Llama-3.1-8B","ibm-granite/granite-3.3-8b-instruct", "Qwen/Qwen3-14B", "mistralai/Mistral-Small-24B-Instruct-2501", "Qwen/Qwen3-32B"]
+
+    for idx, model in enumerate(models):
+        benchmark_name = "scenario2"
+        remote_exp_folder = f"{time.strftime('%Y%m%d-%H%M%S')}_{benchmark_name}"
+        for mode in modes:
+            run_experiment(model, mode, remote_exp_folder)
+        if idx < 4:
+            time.sleep(600)
+        elif idx >=4 and idx <=7:
+            time.sleep(1200)
+        else:
+            time.sleep(1800)
```

Comment:
A bit confused here. You're sleeping some time for larger models. Why?

Reply:
This is to make sure we don't launch a 100 jobs at the same time. We have around 15 models, each with around 10 experiments (chunk size variations). I sleep between models (and adjust to smaller sleep times for smaller models) to make sure we don't overwhelm the cluster.